### ISSUE TITLE
Migration table is already locked

### ISSUE THREAD

**Issue by rafaelrabaco** [deliverr/knex-snowflake-dialect#5]
Hello, when performing the migration, I always get this error.
Would you help me?
![knex-snowflake-dialect-migration](https://user-images.githubusercontent.com/11164992/78154974-463fac00-7413-11ea-86d1-4874b2f1136a.png)
![migrations_lock-table](https://user-images.githubusercontent.com/11164992/78155223-8f8ffb80-7413-11ea-8b6b-e91951a90ef6.png)

**Comment by emurphy** [comment]
Have you tried deleting the `SEC_MIGRATIONS_LOCK` rows as suggested?
What version of knex-snowflake-dialect are you using?

**Comment by rafaelrabaco** [comment]
> Have you tried deleting the `SEC_MIGRATIONS_LOCK` rows as suggested?
> What version of knex-snowflake-dialect are you using?

Yes, I tried to delete the row from the `SEC_MIGRATIONS_LOCK` table
I'm using the latest version `0.2.2`

**Comment by emurphy** [comment]
For some reason the result of `update "SEC_MIGRATIONS_LOCK" set "IS_LOCKED" = 1 where "IS_LOCKED" = 0` is failing the `rowCount` check from [knex/lib/migrate/Migrator.js](https://github.com/knex/knex/blob/master/lib/migrate/Migrator.js#L311) below. What happens when you run the SQL update manually?

```javascript
  _lockMigrations(trx) {
    const tableName = getLockTableName(this.config.tableName);
    return getTable(this.knex, tableName, this.config.schemaName)
      .transacting(trx)
      .where('is_locked', '=', 0)
      .update({ is_locked: 1 })
      .then((rowCount) => {
        if (rowCount != 1) {
          throw new Error('Migration table is already locked');
        }
      });
  }
```

Also, what knex version are you using?

Will help if you can share a minimal repo or test case to reproduce the error.

**Comment by rafaelrabaco** [comment]
> For some reason the result of `update "SEC_MIGRATIONS_LOCK" set "IS_LOCKED" = 1 where "IS_LOCKED" = 0` is failing the `rowCount` check from [knex/lib/migrate/Migrator.js](https://github.com/knex/knex/blob/master/lib/migrate/Migrator.js#L311) below. What happens when you run the SQL update manually?
> 
> ```js
>   _lockMigrations(trx) {
>     const tableName = getLockTableName(this.config.tableName);
>     return getTable(this.knex, tableName, this.config.schemaName)
>       .transacting(trx)
>       .where('is_locked', '=', 0)
>       .update({ is_locked: 1 })
>       .then((rowCount) => {
>         if (rowCount != 1) {
>           throw new Error('Migration table is already locked');
>         }
>       });
>   }
> ```
> 
> Also, what knex version are you using?
> 
> Will help if you can share a minimal repo or test case to reproduce the error.


I put a `console.log` to understand what was going in that function

That was the result
![Screen Shot 2020-04-02 at 10 52 50](https://user-images.githubusercontent.com/11164992/78257673-8d3fa700-74d0-11ea-9714-a9abf3e51fa5.png)

I'm using the latest version of knex `0.20.13`
I will upload a repository soon

**Comment by rafaelrabaco** [comment]
Repository for testing: https://github.com/rafaelrabaco/snowflake-demo-issue

**Comment by emurphy** [comment]
Thank you, @rafaelrabaco! The problem is in [SnowflakeDialect#processResponse](https://github.com/deliverr/knex-snowflake-dialect/blob/master/src/index.ts#L186), and I should have time to fix it early next week.

**Comment by emurphy** [comment]
Fixed in 0.2.3.
